File: Introduction to Computer Science/Week 2 - Array/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdbool.h>
#include <math.h>

int main(void)
{
    // get input text
    string text = get_string("Text: ");

    // declare progarm variables
    int letters = 0;
    int words = 0;
    bool counting = true;
    int sentences = 0;

    // loop for each character in text
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // check for letter
        if (isalpha(text[i]))
        {
            letters++;
        }

        // check for words
        if (text[i] != 32)
        {
            if (counting)
            {
                words++;
                counting = false;
            }
        }
        // if character is space, count again
        else
        {
            counting = true;
        }
        // check for sentences
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            sentences++;
        }
    }

    // calculating average letters per 100 words
    float L = (float)letters * 100 / (float) words;
    // calculating average sentences per 100 words
    float S = (float)sentences * 100 / (float) words;

    // Coleman-Liau index
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // output grade depend on index
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}

File: Introduction to Computer Science/Week 6 - Python/readability/readability.py
from cs50 import get_string


# get input text
text = get_string("Text: ")

# declare program variables
letters = 0
words = 0
sentences = 0
count = True

# loop for each character in text
for i in text:

    # check for letter
    if i.isalpha():
        letters += 1

    # check for words
    if i != " ":
        if count:
            words += 1
            count = False

    # if character is space, count again
    else:
        count = True

    # check for sentences
    if i == "." or i == "!" or i == "?":
        sentences += 1


# calculating average letters per 100 words
L = letters * 100 / words

# calculating average sentences per 100 words
S = sentences * 100 / words

# Coleman-Liau index
index = round(0.0588 * L - 0.296 * S - 15.8)

# output grade depend on index
if (index < 1):
    print("Before Grade 1")

elif (index >= 16):
    print("Grade 16+")

else:
    print(f"Grade {index}")